iOS/DuckDuckGo/Subscription/UserScripts/SubscriptionPagesUseSubscriptionFeature.swift

```diff
@@ -350,17 +362,24 @@ final class DefaultSubscriptionPagesUseSubscriptionFeature: SubscriptionPagesUse
 
         setTransactionStatus(.polling)
 
+        var subscriptionParameters: [String: String]?
+
         // Free Trials Experiment Parameters & Pixels
-        var freeTrialParameters: [String: String]?
         if shouldPerformFreeTrialPostPurchaseActions {
-            freeTrialParameters = completeSubscriptionFreeTrialParameters
+            subscriptionParameters = completeSubscriptionFreeTrialParameters
             fireFreeTrialSubscriptionPurchasePixel(for: subscriptionSelection.id)
         }
 
+        if let experiment = subscriptionSelection.experiment {
+            var parameters = subscriptionParameters ?? [String: String]()
+            experiment.map(to: &parameters)
+            subscriptionParameters = parameters
+        }
```

Comment:
If we have an experiment in the subscription selection we pass the name and cohort as additional parameters.